binrw/src/binread/impls.rs

```diff
@@ -143,14 +255,15 @@ where
         endian: Endian,
         args: Self::Args<'_>,
     ) -> BinResult<Self> {
-        crate::helpers::count_with(args.count, B::read_options)(reader, endian, args.inner)
+        B::read_options_count(reader, endian, args.inner, args.count)
     }
 }
 
 impl<B, const N: usize> BinRead for [B; N]
 where
     B: BinRead,
     for<'a> B::Args<'a>: Clone,
+    [B; N]: for<'b> TryFrom<&'b [B]>,
```

Comment:
does this bound cause any weird backwards compatibility issues? is there a matching blanket implementation in libstd that makes this fine or something?

Reply:
yeah that's the thing that broke in CI, where the fix is to add a Clone bound. It's bad, we need to take a harder look at the api surface of `read_options_count` if we want to support the arrays.

Reply:
The breaking change of relying on Clone has been resolved by using `array_init::from_iter`, not sure what to do about no_std yet. All the current forms of the specializations assume that Vec is in play, the generic form for everything except u8 even more so.

I guess all the optimizations were previously gated behind std, because they were gated behind Vec, eh? I guess that would be one solution, gating behind std, but that kinda sucks.